benchmark/url/new-url-parse.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+const common = require('../common.js');
+const url = require('url');
+const v8 = require('v8');
+
+const bench = common.createBenchmark(main, {
+  type: 'one two three four five'.split(' '),
+  method: ['old', 'new'],
+  n: [25e4]
+});
+
+function useOld(n, input) {
+  // Force-optimize url.parse() so that the benchmark doesn't get
+  // disrupted by the optimizer kicking in halfway through.
+  url.parse(input);
+  v8.setFlagsFromString('--allow_natives_syntax');
+  eval('%OptimizeFunctionOnNextCall(url.parse)');
+
+  bench.start();
+  for (var i = 0; i < n; i += 1)
+    url.parse(input);
+  bench.end(n);
+}
+
+function useNew(n, input) {
+  bench.start();
+  for (var i = 0; i < n; i += 1)
+    new url.URL(input);
+  bench.end(n);
+}
+
+function main(conf) {
+  const type = conf.type;
+  const n = conf.n | 0;
+  const method = conf.method;
+
+  var inputs = {
+    one: 'http://nodejs.org/docs/latest/api/url.html#url_url_format_urlobj',
+    two: 'http://blog.nodejs.org/',
+    three: 'https://encrypted.google.com/search?q=url&q=site:npmjs.org&hl=en',
+    four: 'javascript:alert("node is awesome");',
+    //five: 'some.ran/dom/url.thing?oh=yes#whoo',
```

Comment:
The initial goal to impl the WHATWG parsing spec as is without introducing
any additional default behavior. As such, since that spec does not assume a
default, neither does this impl. I do believe it may be worthwhile adding
an escape hatch for extended behaviors with the understanding that those
are nonstandard.

On Jul 20, 2016 10:09 AM, "Ingvar Stepanyan" <notifications@github.com>
wrote:

> In benchmark/url/new-url-parse.js
> <https://github.com/nodejs/node/pull/7448#discussion_r71566061>:
>
> > +  for (var i = 0; i < n; i += 1)
> > +    new url.URL(input);
> > +  bench.end(n);
> > +}
> > +
> > +function main(conf) {
> > +  const type = conf.type;
> > +  const n = conf.n | 0;
> > +  const method = conf.method;
> > +
> > +  var inputs = {
> > +    one: 'http://nodejs.org/docs/latest/api/url.html#url_url_format_urlobj',
> > +    two: 'http://blog.nodejs.org/',
> > +    three: 'https://encrypted.google.com/search?q=url&q=site:npmjs.org&hl=en',
> > +    four: 'javascript:alert("node is awesome");',
> > +    //five: 'some.ran/dom/url.thing?oh=yes#whoo',
>
> Perhaps use http:// as default base URL for the parser?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/nodejs/node/pull/7448/files/367924f52d393fc398a340bb5ee60d1a10a5cdf2#r71566061>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAa2ecJo9qtJYSMQ_zYGpiq-kjQwd8sSks5qXlZPgaJpZM4I_c8x>
> .
>